docs/fastdds/transport/zero_copy/zero_copy.rst

```diff
@@ -0,0 +1,161 @@
+
+Writing and reading in Zero-Copy transfers
+------------------------------------------
+
+The following is an example of how to publish and receive samples with DataWriters and DataReaders respectively
+that implement Zero-Copy.
+
+DataWriter
+^^^^^^^^^^
+
+When the DataWriter is created, *Fast DDS* will pre-allocate a pool of
+|ResourceLimitsQosPolicy::max_samples-api| + ``extra_samples`` samples that reside in a shared memory mapped file.
+This poll will be used to loan samples when the :func:`loan_sample` function is called.
+
+An application example of a DataWriter that supports Zero-Copy using the *Fast DDS* library is presented below.
+There are several points to note in the following code:
+
+*   Enabling the ``DataSharingQosPolicy``.
+*   The use of the :func:`return_loan` function to access and modify data samples.
+*   The writing of data samples.
+
+.. literalinclude:: ../../../../code/DDSCodeTester.cpp
+   :language: c++
+   :dedent: 8
+   :start-after: //LOANABLE_HELLOWORLD_EXAMPLE_WRITER
+   :end-before: //!
+
+
+DataReader
+^^^^^^^^^^
+
+The following is an application example of a DataWriter that supports Zero-Copy using the *Fast DDS* library.
+As shown in this code snippet, the only difference in the DataReader declaration is enabling the
+|DataReaderQoS| ``DataSharingQosPolicy``.
+
+.. literalinclude:: ../../../../code/DDSCodeTester.cpp
+   :language: c++
+   :dedent: 8
+   :start-after: //LOANABLE_HELLOWORLD_EXAMPLE_READER
+   :end-before: //!
+
+Finally, the code snippet below implements the |DataReaderListener::on_data_available-api| |DataReaderListener|
+callback.
+The key points to be noted in this function are:
+
+*   The declaration and handling of ``LoanableSequences``.
+*   The use of the :func:`return_loan` function to indicate to the DataReader that the application has finished
+    accessing the sequence.
+
+.. literalinclude:: ../../../../code/DDSCodeTester.cpp
+   :language: c++
+   :dedent: 12
+   :start-after: //LOANABLE_HELLOWORLD_EXAMPLE_LISTENER_READER
+   :end-before: //!
+
+Caveats
+-------
+
+*   After calling :func:`loan_sample`, Fast DDS takes ownership of the sample and therefore it is no longer safe
+    to make changes to that sample.
+*   If function :func:`loan_sample` is called first and the sample is never written, it is necessary to use function
+    :func:`discard_loan` to return the sample to the DataWriter.
+    If this is not done, the subsequent calls to :func:`loan_sample` may fail if DataWriter has no more
+    ``extra_samples`` to loan.
+*   The current maximum supported sample size is the maximum value of an ``uint32_t``.
+
+Constraints
+-----------
+
+Although Zero-Copy can be used for one or several *Fast DDS* application processes running on the same machine,
+it has some constraints:
+
+*   Complete support for plain types.
+*   Unbounded types are not supported.
+*   Suitable for |PREALLOCATED_MEMORY_MODE-api| memory configurations only.
```

Comment:
I think it also works for `PREALOCATED_WITH_REALLOC`, as bounded plain types will not yield to reallocation.

Reply:
:+1: